server/src/main/java/org/elasticsearch/rest/RestResponse.java

```diff
@@ -115,7 +115,7 @@ public RestResponse(RestChannel channel, Exception e) throws IOException {
     @SuppressWarnings("this-escape")
     public RestResponse(RestChannel channel, RestStatus status, Exception e) throws IOException {
         this.status = status;
-        ToXContent.Params params = paramsFromRequest(channel.request());
+        ToXContent.Params params = channel.request();
         if (params.paramAsBoolean(REST_EXCEPTION_SKIP_STACK_TRACE, REST_EXCEPTION_SKIP_STACK_TRACE_DEFAULT) && e != null) {
```

Comment:
Do we need still need this check here?

```
params.paramAsBoolean(REST_EXCEPTION_SKIP_STACK_TRACE, REST_EXCEPTION_SKIP_STACK_TRACE_DEFAULT) 
```

Do we want to check if only if `status.getStatus() < 500` ?

Reply:
Reading this again more closely, REST_EXCEPTION_SKIP_STACK_TRACE is only ever set to "false" in GetFeatureUpgradeStatusResponse, ExceptionWatchRecord and ILMHistoryItem because it seems someone decided to always render exception stack traces inside those objects. That also means it should never be set here, so you are right this check can be removed. The presence of an exception is still something we'd need to check I think, also we want to log both 400/500s here so I don't see why we should condition on status here. I will push some changes.